src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,221 @@
+  *                       the rock volume of interest. The inputs in each array are strike, dip and the x,
+  *                       y and z coordinates of a point that lies within the plane containing the face.
+  *                       Strike and dip should be specified in degrees.
+  * @param jointSetData Array of arrays containing the input data representing joint sets. The inputs in each array
+  *                     are strike, dip, joint spacing, persistence, phi, cohesion and optional stochastic parameters
+  */
+case class JointGenerator(globalOrigin: Array[Double], boundingBox: Array[Double], rockVolumeData: Seq[Array[Double]],
+                     jointSetData: Seq[Array[Double]]) {
+  val origin = globalOrigin
+  val lowerLeftCorner = Array(boundingBox(0), boundingBox(1), boundingBox(2))
+  val upperRightCorner = Array(boundingBox(3), boundingBox(4), boundingBox(5))
+  val rockVolume = JointGenerator.findBoundingFaces(globalOrigin, rockVolumeData)
+  val masterJoints = JointGenerator.findMasterJoints(globalOrigin, lowerLeftCorner, jointSetData).get
+
+  if (masterJoints.isEmpty) {
+    throw new IllegalArgumentException("ERROR: Stochastic joint generation not yet implemented")
+  }
+
+  val jointSets = generateJointSets(jointSetData)
+
+  /**
+    * Generates joint sets for each of the input joint sets
+    *
+    * @param jointSetData Array of arrays containing the input data representing joint sets. The inputs in each array
+    *                     are strike, dip, joint spacing, persistence, phi, cohesion and optional stochastic parameters
+    * @return Seq of joints for each input joint set
+    */
+  def generateJointSets(jointSetData: Seq[Array[Double]]): Seq[Seq[Joint]] = {
+    val jointTuple = masterJoints.zip(jointSetData)
+    
+    val deltaX = upperRightCorner(0) - lowerLeftCorner(0)
+    val deltaY = upperRightCorner(1) - lowerLeftCorner(1)
+    val deltaZ = upperRightCorner(2) - lowerLeftCorner(2)
+
+    val lowerCorners = Array(lowerLeftCorner,
+      Array(lowerLeftCorner(0) + deltaX, lowerLeftCorner(1), lowerLeftCorner(2)),
+      Array(lowerLeftCorner(0) + deltaX, lowerLeftCorner(1) + deltaY, lowerLeftCorner(2)),
+      Array(lowerLeftCorner(0), lowerLeftCorner(1) + deltaY, lowerLeftCorner(2))
+    )
+
+    val upperCorners = Array(
+      Array(lowerCorners(2)(0), lowerCorners(2)(1), lowerCorners(2)(2) + deltaZ),
+      Array(lowerCorners(3)(0), lowerCorners(3)(1), lowerCorners(3)(2) + deltaZ),
+      Array(lowerCorners(0)(0), lowerCorners(0)(1), lowerCorners(0)(2) + deltaZ),
+      Array(lowerCorners(1)(0), lowerCorners(1)(1), lowerCorners(1)(2) + deltaZ)
+    )
+
+    // Diagonal vectors of bounding box
+    val diagonalVectors = upperCorners.zip(lowerCorners) map { case (upper, lower) =>
+        linalg.normalize(DenseVector[Double](upper(0) - lower(0), upper(1) - lower(1), upper(2) - lower(2)))
+    }
+
+    jointTuple map { case (joint, jointData) =>
+      if (jointData(3) == 100) {
```

Comment:
All these arrays to represent complex data types is confusing. :)

I have some ideas on how we can improve this I'll add in a comment on the overall pull request.